drush.complete.sh

```diff
@@ -24,9 +24,40 @@ __drush_ps1() {
     __DRUPAL_SITE="$DRUPAL_SITE"
   fi
 
+  if [ -n "${__DRUPAL_SITE}" ] && [ -n "${DRUSH_PS1_SHOWCOLORHINTS-}" ]; then
+    __drush_ps1_colorize
+  fi
+
   [[ -n "$__DRUPAL_SITE" ]] && printf "${1:- (%s)}" "$__DRUPAL_SITE"
 }
 
+# Helper function meant to be called from __drush_ps1.
+__drush_ps1_colorize() {
+  if [[ -n ${ZSH_VERSION-} ]]; then
+    local red='%F{red}'
+    local green='%F{green}'
+    local blue='%F{blue}'
+    local clear='%f'
+  else
+    # Using \[ and \] around colors is necessary to prevent issues with command
+    # line editing/browsing/completion!
```

Comment:
I stole these color definitions from [git/git-prompt.sh](https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh), along with this comment, and I'm just taking their word for it.